Shared/messages/MessageFeed.cpp

```diff
@@ -17,44 +17,156 @@
 // PCH header
 #include "pch.hpp"
 
-#include "MessageFeed.h"
+// C++ API Headers
+#include "Error.h"
+#include "Field.h"
+#include "Domain.h"
+#include "Renderer.h"
+#include "SymbolTypes.h"
+#include "ServiceTypes.h"
+#include "SpatialReference.h"
+#include "DynamicEntity.h"
+#include "DynamicEntityInfo.h"
+#include "DynamicEntityLayer.h"
+#include "DynamicEntityDataSourceInfo.h"
+#include "DynamicEntityObservation.h"
+#include "DynamicEntityObservationInfo.h"
+#include "AttributeListModel.h"
 
-// dsa app headers
+// DSA app headers
+#include "Message.h"
+#include "MessageFeed.h"
 #include "MessagesOverlay.h"
 
+using namespace Esri::ArcGISRuntime;
+
 namespace Dsa {
 
 /*!
   \class Dsa::MessageFeed
   \inmodule Dsa
-  \inherits QObject
+  \inherits DynamicEntityDataSource
   \brief Represents a feed for a given message type which will be displayed on a
   \l MessageOverlay.
  */
 
 /*!
-  \brief Constructor accepting an optional \a parent.
+  \brief Constructor accepting a feed \a name, a message \a type, and \a overlay and an optional \a parent.
  */
-MessageFeed::MessageFeed(QObject* parent) :
-  QObject(parent)
+MessageFeed::MessageFeed(const QString& name, const QString& type, QObject* parent) :
+  DynamicEntityDataSource(parent),
+  m_feedName(name),
+  m_feedMessageType(type)
 {
 }
 
-/*!
-  \brief Constructor accepting a feed \a name, a message \a type, and \a overlay and an optional \a parent.
- */
-MessageFeed::MessageFeed(const QString& name, const QString& type, MessagesOverlay* overlay, QObject* parent) :
-  QObject(parent),
-  m_feedName(name),
-  m_feedMessageType(type),
-  m_messagesOverlay(overlay)
+MessageFeed::~MessageFeed()
```

Comment:
For any destructors that don't do anything, you can do `= default;` in the .cpp files.

Reply:
Fixed.